pkg/grafana/provider.go

```diff
@@ -1,41 +1,105 @@
 package grafana
 
 import (
+	"fmt"
+	"net/url"
 	"path/filepath"
 
 	gclient "github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grizzly/pkg/config"
 	"github.com/grafana/grizzly/pkg/grizzly"
 )
 
 // Provider is a grizzly.Provider implementation for Grafana.
 type Provider struct {
-	client *gclient.GrafanaHTTPAPI
+	context *config.Context
+	client  *gclient.GrafanaHTTPAPI
+}
+
+type ClientProvider interface {
+	Client() (*gclient.GrafanaHTTPAPI, error)
+	Current() (*config.Context, error)
 }
 
 // NewProvider instantiates a new Provider.
-func NewProvider(client *gclient.GrafanaHTTPAPI) *Provider {
+func NewProvider() *Provider {
+	return &Provider{}
+}
+
+// NewProviderWithConfig instantiates a new Provider.
+func NewProviderWithConfig(conf *config.Config) *Provider {
 	return &Provider{
-		client: client,
+		context: conf.Current(),
 	}
 }
 
 // Group returns the group name of the Grafana provider
-func (p Provider) Group() string {
+func (p *Provider) Group() string {
 	return "grizzly.grafana.com"
 }
 
 // Version returns the version of this provider
-func (p Provider) Version() string {
+func (p *Provider) Version() string {
 	return "v1alpha1"
 }
 
+func (p *Provider) Current() (*config.Context, error) {
+	if p.context == nil {
+		exists, err := config.Exists()
+		if err != nil {
+			return nil, fmt.Errorf("Error locating configuration file: %v", err)
+		}
+		if exists {
+			conf, err := config.Load()
+			if err != nil {
+				return nil, err
+			}
+			p.context = conf.Current()
+		} else {
+			conf, err := config.FromEnvironment()
+			if err != nil {
+				return nil, err
+			}
+			p.context = conf.Current()
+		}
+	}
+	return p.context, nil
+}
+
+func (p *Provider) Client() (*gclient.GrafanaHTTPAPI, error) {
+	if p.client != nil {
+		return p.client, nil
+	}
+
+	_, err := p.Current()
+	if err != nil {
+		return nil, err
+	}
+	parsedUrl, err := url.Parse(p.context.Grafana.URL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid Grafana URL")
+	}
+
+	transportConfig := gclient.DefaultTransportConfig().WithHost(parsedUrl.Host).WithSchemes([]string{parsedUrl.Scheme})
```

Comment:
nit: A few linebreaks haven't hurt anyone.

Reply:
You clearly never coded in Java...